[Fan Haolin] iP

kaoxi998533 opened the pull request:
# D++e

**D++e** is a *homework project* created by ***a student***. It has the following advantage:
- Runs completely on CLI, friendly to users who type fast
- Neat user interface

## Setting up in Intellij
The code is written in Java and you can edit it if you want. Here is a sample code:
```Java
 public Task(String description) {
        this.description = description;
        this.isDone = false;
    }
```
With this code, `Task("hi")` creates a new task.
- [X] It is open-source
- [X] It still has room for improvement
- [ ] It has you involved as a member of project! :sweat_smile:

> To code, or not to code, that is the question --- me

[If you like my project, then click here!](https://github.com/nus-cs2103-AY2425S1/ip)

Prerequisites: JDK 17, update Intellij to the most recent version.

1. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project first)
1. Open the project into Intellij as follows:
   1. Click `Open`.
   1. Select the project directory, and click `OK`.
   1. If there are any further prompts, accept the defaults.
1. Configure the project to use **JDK 17** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
   In the same dialog, set the **Project language level** field to the `SDK default` option.
3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
   ```
   Hello from
    ____        _        
   |  _ \ _   _| | _____ 
   | | | | | | | |/ / _ \
   | |_| | |_| |   <  __/
   |____/ \__,_|_|\_\___|
   ```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

kaoxi998533 committed:
added function for adding horizontal lines

kaoxi998533 committed:
Level 0 done

kaoxi998533 committed:
added echo function

kaoxi998533 committed:
add and list are done

kaoxi998533 committed:
Task classes done. Mark as done is also done.

kaoxi998533 committed:
added different types of tasks

kaoxi998533 committed:
Added exceptions and try-catch statement

kaoxi998533 committed:
added delete item in list feature and updated test

kaoxi998533 committed:
Add skeleton for parser class

kaoxi998533 committed:
Add Command and its implementing classes and TaskList class

kaoxi998533 committed:
Adjust handling of file that saves todo list

kaoxi998533 committed:
Implement UI class

kaoxi998533 committed:
Integrate most logic into Duke class

kaoxi998533 committed:
Fix the issue of scanner not closed

kaoxi998533 committed:
Merge branch 'branch-A-MoreOOP'

kaoxi998533 committed:
Organize classes into packages

kaoxi998533 committed:
Merge branch 'branch-A-Packages'

kaoxi998533 committed:
Fix bug that occur on running

kaoxi998533 committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-A-Gradle

kaoxi998533 committed:
Add gradle

shotnothing commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,26 @@
+        this.taskList = taskList;
+    }
+
+    @Override
+    public void execute() {
```
missing javadoc!

shotnothing commented on src/main/java/utilities/Parser.java:
```diff
@@ -0,0 +1,77 @@
+public class Parser {
+    public static String addHorizontalLinesAndIndentation(String dialog) {
+        StringBuilder res = new StringBuilder("    ____________________________________________________________\n");
+        Scanner sc = new Scanner(dialog);
+        while (sc.hasNextLine()) {
+            String line = sc.nextLine();
+            res.append("      ").append(line.trim()).append("\n"); // Ensure consistent indentation
+        }
+        res.append("    ____________________________________________________________");
+        sc.close();
```
might be more efficient to persist scanner across method calls by keeping the scanner open as a property?

shotnothing commented on src/main/java/utilities/Ui.java:
```diff
@@ -0,0 +1,78 @@
+        this.taskList = taskList;
+    }
+
+    public void printWelcomeMessage() {
+        String hi = "Hello! I'm Foo\n" +
+                "What can I do for you?";
+        System.out.println(Parser.addHorizontalLinesAndIndentation(hi));
+    }
+
+    public void printGoodbyeMessage() {
+        System.out.println(Parser
+            .addHorizontalLinesAndIndentation(
+                "Bye. Hope to see you again soon!"));
+    }
```
i think stuff like this can be static

kaoxi998533 committed:
Add test for TaskList and MarkCommand

kaoxi998533 committed:
Modify build.gradle file for building jar

kaoxi998533 committed:
Merge branch 'branch-A-JUnit'

kaoxi998533 committed:
Add JavaDoc for most methods and classes

kaoxi998533 committed:
Adjust some poor practices of code

kaoxi998533 committed:
Add find functions

kaoxi998533 committed:
Merge branch 'branch-Level-9'

kaoxi998533 committed:
Merge branch 'branch-A-CodingStandard'

kaoxi998533 committed:
Solve conflict between Javadoc branch and master branch

Jianwen0451 commented on src/main/java/command/MarkCommand.java:
```diff
@@ -0,0 +1,32 @@
+    public void execute() {
+        taskList.markTaskAsDone(index - 1);
+        new Storage("data/duke.txt").save(taskList);
+        System.out.println(Parser.addHorizontalLinesAndIndentation("Nice! I've marked this task as done:\n" + taskList.get(index - 1)));
```
Maybe be can separate this line into two shorter lines?

Jianwen0451 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,232 @@
+
+        Task task;
+        switch (type) {
+            case "T":
```
There should be no indentation for "case". Maybe change it to the same indentation as switch keyword?

Jianwen0451 commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,37 @@
+ * DeleteCommand class is used to delete a task from the task list.
+ */
+public class DeleteCommand extends Command {
+    private int index;
```
I like the way you name your Commands and its execute method! We actually name them in the same way!

Jianwen0451 commented on src/main/java/command/ListCommand.java:
```diff
@@ -0,0 +1,38 @@
+package command;
+import task.Task;
```
Maybe a separate line between packages and import?

Jianwen0451 commented on src/test/java/command/MarkCommandTest.java:
```diff
@@ -0,0 +1,39 @@
+package command;
```
I like it that you even form different packages for test cases! That is a good practice!

Jianwen0451 commented on src/main/java/exception/DukeException.class:
Maybe add all these .class binary file to .gitignore?

Jianwen0451 reviewed (commented):
Generally a nice piece of work! And I comment on some places for you to improve!

kaoxi998533 committed:
Update README.md

kaoxi998533 committed:
Changed name for the project

kaoxi998533 committed:
Merge branch 'master' of https://github.com/kaoxi998533/ip

kaoxi998533 committed:
Fix bug

ethanwangkangen commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,232 @@
+    public static Task fromFileString(String s) {
+        String[] parts = s.split("\\|");
+        String type = parts[0];
+        String isDone = parts[1];
```
Maybe change this variable name so it isn't confused with the Boolean isDone of Task?

ethanwangkangen commented on src/main/java/utilities/Parser.java:
```diff
@@ -0,0 +1,103 @@
+     * @param dialog The dialog to be formatted.
+     * @return The formatted dialog.
+     */
+    public static String addHorizontalLinesAndIndentation(String dialog) {
```
I'm not sure if this function name is considered too long under "code quality" rules. Something shorter could be a name like "formatDialog()" but that would miss out on explaining what the function does explicitly.

ethanwangkangen commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,232 @@
+
+        @Override
+        public String toString() {
+            return String.format("[T][%s] %s (from: %s to: %s)",
```
Should this be [E] for event instead?

ethanwangkangen commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,232 @@
+     * Method setIsDone is used to set the status of the task.
+     * @param val The status of the task.
+     */
+    public void setIsDone(boolean val) {
```
Good that you follow naming conventions for setters and getters!

ethanwangkangen reviewed (commented):
LGTM overall!

kaoxi998533 committed:
Complete UI changes

kaoxi998533 committed:
Merge branch 'branch-Level-10'

kaoxi998533 committed:
Assertions are added to parts of the code that
have assumptions.

This allows easier debugging for future.

Let's have `assert` keywords at places where it is
needed.

This detects logical errors at these places that are
deemed important and error-prone.

kaoxi998533 committed:
Parser, Task: improve code quality.

The code quality was poor as it was lengthy ang against SLAP.

Refactor the code and extract the sections out as methods to
ensure conciseness of code and that all codes in the same function
are of the same abstraction layer.

This will make more readable and flexible to future changes.

kaoxi998533 committed:
Merge pull request #1 from kaoxi998533/branch-A-Assertions

Assertions are added to parts of the code that

kaoxi998533 committed:
Merge pull request #2 from kaoxi998533/branch-A-CodeQuality

Parser, Task: improve code quality.

kaoxi998533 committed:
Add task priority sorting functionality

kaoxi998533 committed:
Merge branch 'branch-C-Sort'

kaoxi998533 committed:
Use final static field for file path

kaoxi998533 committed:
Change name of text file for storing data

kaoxi998533 committed:
Change name of text file to store data and corresponding codes

kaoxi998533 committed:
Ensure all parts of code do not contain Duke

kaoxi998533 committed:
Improve error handling

kaoxi998533 committed:
Fix style

kaoxi998533 committed:
Update markdown file

kaoxi998533 committed:
Update markdown file

kaoxi998533 committed:
Update markdown file

kaoxi998533 committed:
Complete project website
